app/markdown/long_running_actions.md

```diff
@@ -0,0 +1,172 @@
+## Managing Long Running Processes with Services
+
+Most *signals* in your application are triggered by users interacting with the UI-- typing in an input, submitting a form, or closing a modal window. But, many times your application needs to respond to events from the "outside" world. These could be websocket messages from services like Firebase, chat messages from your backend, or even simple HTTP long polling for data.
+
+In Cerebral, these processes are encapsulated using the *module* interface and are exposed as *services* to your actions. Let's walk through an example HTTP long polling service.
+
+#### An HTTP Polling Module
+
+Imagine you have a Navigation bar at the top of your app that needs to display the unread message count for the current user. It queries an `/api/message_counts/:id` endpoint every 10 seconds to fetch the current `newMessageCount` for the user and display in a component.
+
+First, let's set up a skeleton with the expected *module* interface. This hooks our code into cerebral and gives us access to the controller's `services`, `state`, and `signals` and also lets us define our own services, signals, and state:
+
+```js
+// ./modules/http-poller/index.js
+
+export default (options = {}) => {
+  return (module, controller) => {
+  }
+}
+```
+The *module* interface is a higher order function returning another function that is passed the cerebral `controller` and the `module` instance as arguments. It optionally accepts configuration `options`.
```

Comment:
HOF not important really... typically only used for shared modules :) Unless options is mentioned again? If so maybe introduce it then instead?